src/OrbitGl/SchedulingStats.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_SCHEDULING_STATS_H_
+#define ORBIT_GL_SCHEDULING_STATS_H_
+
+#include <functional>
+#include <map>
+#include <string>
+#include <vector>
+
+class CaptureData;
+class SchedulerTrack;
+class TextBox;
+
+// SchedulingStats is a utility class to generate statistics about system-wide scheduling
+// information. This includes core occupancy information as well as system-wide process and thread
+// statistics.
+
+class SchedulingStats {
+ public:
+  typedef std::function<std::string(int32_t)> ThreadNameProvider;
+
+  SchedulingStats() = delete;
+  SchedulingStats(const std::vector<const TextBox*>& scheduling_scopes,
+                  const ThreadNameProvider& thread_name_provider, uint64_t start_ns,
+                  uint64_t end_ns);
+
+  [[nodiscard]] std::string ToString() const;
+
+  struct ThreadStats {
+    int32_t tid = -1;
+    uint64_t time_on_core_ns_ = 0;
+    double time_on_core_ms = 0;
+    std::string thread_name;
+  };
+
+  struct ProcessStats {
+    std::map<int32_t, ThreadStats> thread_stats_by_tid;
+    std::vector<ThreadStats*> thread_stats_sorted_by_time_on_core;
+    int32_t pid = -1;
+    uint64_t time_on_core_ns_ = 0;
+    double time_on_core_ms = 0;
+    std::string process_name;
+  };
+
+  struct ProcessStatsTimeSorter {
+    bool operator()(const ProcessStats* a, const ProcessStats* b) const {
+      return a->time_on_core_ns_ > b->time_on_core_ns_;
+    }
+  };
+
+  struct ThreadStatsTimeSorter {
+    bool operator()(const ThreadStats* a, const ThreadStats* b) const {
+      return a->time_on_core_ns_ > b->time_on_core_ns_;
+    }
+  };
+
+  double GetTimeRangeMs() const { return time_range_ms_; }
+  uint64_t GetTimeOnCoreNs() const { return time_on_core_ns_; }
+  const std::map<int32_t, uint64_t>& GetTimeOnCoreNsByCore() const {
+    return time_on_core_ns_by_core_;
+  }
+  const std::map<int32_t, ProcessStats>& GetProcessStatsByPid() const {
```

Comment:
I would expect this function, as well as the previous one, to be more like 
`  const ProcessStats& GetProcessStats(int32_t pid) const {
`
but feel free to disregard this comment

If you make this change, _GetTimeOnCoreNsByCore_ could be called _GetTimeOnCoreNs_

Reply:
Oh, yeah this needs to return the map as the map contains information about the number of processes that we have stats for and this is used in the tests. I prefer leaving as is, but thanks for the comment.